### ISSUE TITLE
Formatting using eslint_local fails

### ISSUE THREAD

**Issue by purushnayak** [Chiel92/vim-autoformat#280]
Error reported is below with verbose enabled.

> Trying definition from g:formatdef_eslint_local
> Evaluated formatprg: node_modules/.bin/eslint -c /Users/user1/dev/js/eslint1/.eslintrc.json --fix /var/folders/lr/8tw32hd16d1c81v8_fvf238h0000gn/T/nvimdXCMsH/10.ts 1> /dev/null; exit_code=$? cat /var/folders/lr/8tw32hd16d1c81v8_fvf238h0000gn/T/nvimdXCMsH/10.ts; exit $exit_code
> Using python 3 code...
> Formatter eslint_local has errors: b'/bin/sh: node_modules/.bin/eslint: No such file or directory\n'
> Definition in 'g:formatdef_eslint_local' was unsuccessful.
> No format definitions were successful.
> Removing trailing whitespace...
> Retabbing...
> Autoindenting...
> 3 lines to indent... 
> 4 lines indented 

Seems like coming from this part of code Running :!node_modules/.bin/eslint works (Need to set something in environ?).

```
    formattername = vim.eval('b:formatters[s:index]')
    if stderrdata:
        if verbose:
            print('Formatter {} has errors: {}'.format(formattername, stderrdata))
```

**Comment by purushnayak** [comment]
I tried prepending getcwd() to the l:prog in default.vim but now I get this error. Not sure how to set the working directory so everything runs from the same place.

> Trying definition from g:formatdef_eslint_local
> Evaluated formatprg: /Users/annu/user1/js/eslint1/node_modules/.bin/eslint -c /Users/user1/dev/js/eslint1/.eslintrc.json --fix /var/folders/lr/8tw32hd16d1c81v8_fvf238h0000gn/T/nvimjDGAWc/6.ts 1> /dev/null; exit_code=$? cat /var/folders/lr/8tw32hd16d1c81v8_fvf238h0000gn/T/nvimjDGAWc/6.ts; rm -f /var/folders/lr/8tw32hd16d1c81v8_fvf238h0000gn/T/nvimjDGAWc/6.ts; exit $exit_code
> Using python 3 code...
> Formatter eslint_local has errors: b'\nOops! Something went wrong! :(\n\nESLint: 6.7.2.\n\nESLint couldn\'t find the plugin "@typescript-eslint/eslint-plugin".\n\n(The package "@typescript-eslint/eslint-plugin" was not found when loaded as a Node module from the directory "/Users/user1".)\n\nIt\'s likely that the plugin isn\'t installed correctly. Try reinstalling by running the following:\n\n    npm install @typescript-eslint/eslint-plugin@latest --save-dev\n\nThe plugin "@typescript-eslint/eslint-plugin" was referenced from the config file in "--config".\n\nIf you still can\'t figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.\n\n'
> Definition in 'g:formatdef_eslint_local' was unsuccessful.
>

**Comment by purushnayak** [comment]
I was able to get this to work using cd: %:h:p with a file open in the root directory of the project. Running the plugin after that with the above fix works. So we just need to append getcwd() to path returned by findfile().

**Comment by Chiel92** [comment]
If you look at defaults.vim line 192 you can see that the eslint logic is horrifyingly complicated. I don't know why eslint makes it so difficult to run itself, but anyway, if you discover any bugs, this is probably the piece of code that needs fixing.

**Comment by purushnayak** [comment]
I just submitted a pull request to just prepend cwd to path of eslint. That works for me. Autoformat should be run with the current directory as the root of the project.

This is a very useful plugin. Thanks for creating/maintaining it.

**Comment by purushnayak** [comment]
> If you look at defaults.vim line 192 you can see that the eslint logic is horrifyingly complicated. I don't know why eslint makes it so difficult to run itself, but anyway, if you discover any bugs, this is probably the piece of code that needs fixing.

Yeah, there is no way to write output to stdout. Could have saved writing to an actual file.
